apps/umpire/app.js

```diff
@@ -0,0 +1,435 @@
+      // calculate over time
+      var overDuration = new Date(
+        timeSig.getTime() - overTimes[overTimes.length - 1]);
+      var overMinutesString = formatDuration(overDuration);  
+      // log end of over
+      addLog(timeSig, over + 1, 0, 
+        /*LANG*/"Over Duration", overMinutesString);
+      addLog(timeSig, over + 1, 0, 
+        /*LANG*/"Innings Duration", matchMinutesString);
+      overTimes.push(timeSig.getTime());
+      // start new over
+      over += 1;
+      counter = 0; 
+      ballTimes = [];
+    }
+  }
+  // refresh in-play screen
+  g.clear(1);
+  // draw wickets fallen (top-right)
+  g.setFontAlign(1,0);
+  g.setFont("Vector",26).
+   drawString(wickets, 162, 14, true);
+  g.setFont("Vector",12).
+   drawString('\¦\¦\¦', 173, 15, true);
+  // draw battery and heart rate (top-left)
+  g.setFontAlign(-1,0);
+  var heartRateString = 'HR:' + heartRate;
+  if(heartRateEventSeconds <= 0) heartRateString = '';
+  g.setFont("Vector",16).
+    drawString(battery + '% ' + heartRateString, 5, 11, true);
+  // draw clock (upper-centre)
+  g.setFontAlign(0,0);
+  g.setFont("Vector",48).
+    drawString(formatTimeOfDay(timeSig), 93, 55, true);
+  // draw over.ball (centre)
+  var ballString = (over-1) + "." + counter;
+  if(over > OVERS_PER_INNINGS) 
+    ballString = 'END';
+  g.setFont("Vector",80).
+    drawString(ballString, 93, 120, true);
+  // draw ball graph and elapsed time
+  var ballGraph = 
+    BALL_FACED_CHAR.repeat(counter) 
+    + BALL_TO_COME_CHAR.repeat(BALLS_PER_OVER - counter);
+  if(timeCalled) ballGraph = '-TIME-';
+  g.setFont("Vector",18).drawString(
+    ballGraph + ' ' + formatDuration(deadDuration), 93, 166, true);
+  // return to wait for next input
+  processing = false;
+}
+
+function resumeGame(play) {
+  processing = true;
+  Bangle.buzz();
+  Bangle.setUI({
+      mode: "custom",
+      swipe: (directionLR, directionUD)=>{
+        if (directionLR==-1) { 
+          processing = true;
+          menu = showMainMenu();
+        } else if (directionLR==1) { 
+          processing = true;
+          menu = showLog();
+        } else if (directionUD==-1) { 
+          processing = true;
+          countDown(1);
+        } else {
+          processing = true;
+          countDown(-1);
+        }
+      },
+      btn: ()=>{
+        processing = true;
+        countDown(1);
+      }
+    });
+  if(over==0) { // at start of innings
+    over += 1; // N.B. 1-based overs in code
+    counter = 0; // balls
+    ballTimes = [];
+    // set an inital time for comparison  
+    var timeSig = new Date();
+    overTimes.push(timeSig.getTime());
+    addLog(timeSig, over, counter, 
+      "Play", "");        
+    // set up twist refresh once only
+    Bangle.on('twist', function() { 
+      if(!processing) {
+        processing = true; // debounce
+        countDown(0);
+      }
+    });
+    // set up HRM listener once only
+    Bangle.on('HRM', function(h) {
```

Comment:
If this is called multiple times it'll add more functions to the hrm event, perhaps you'll want to remove the previous listener first